server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -83,92 +123,149 @@ public void beforeRefresh() throws IOException {
     /**
      * Upload new segment files created as part of the last refresh to the remote segment store.
      * This method also uploads remote_segments_metadata file which contains metadata of each segment file uploaded.
+     *
      * @param didRefresh true if the refresh opened a new reference
      */
     @Override
     public void afterRefresh(boolean didRefresh) {
-        synchronized (this) {
-            try {
-                if (indexShard.getReplicationTracker().isPrimaryMode()) {
-                    if (this.primaryTerm != indexShard.getOperationPrimaryTerm()) {
-                        this.primaryTerm = indexShard.getOperationPrimaryTerm();
-                        this.remoteDirectory.init();
+        try {
+            indexShard.getThreadPool().executor(ThreadPool.Names.REMOTE_REFRESH).submit(() -> syncSegments(false)).get();
+        } catch (InterruptedException | ExecutionException e) {
+            logger.info("Exception occurred while scheduling syncSegments", e);
+        }
+    }
+
+    private synchronized void syncSegments(boolean retry) {
```

Comment:
nit: `isRetry` makes it more clear similar to how you added it in `afterSegmentsSync`